### PR TITLE
DefaultRenderingPipeline to webgpu

### PR DESCRIPTION
**Pull request by deltakosh**
I flag this one as breaking change because now the PostProcess class will not be able to returns the effect with `getEffect` right away. As the system needs to load the shader code for the engine, the users will have to wait for `postProcess.onEffectCreatedObservable` to be able to get the effect.

### CODE REVIEW

**Review by sebavan — APPROVED** [review]

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15398/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15398/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15398/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15398/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15398/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15398/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15398/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15398/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15398/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGL 1 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15398/merge/testResults/webgl1/index.html

If tests were successful afterwards, this report might not be available anymore.

File: packages/dev/core/src/ShadersWGSL/anaglyph.fragment.fx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+
+#define CUSTOM_FRAGMENT_DEFINITIONS
+
+fn main(void)
+{
+    var leftFrag: vec4f = texture2D(leftSampler, vUV);
+    leftFrag =  vec4f(1.0, leftFrag.g, leftFrag.b, 1.0);
+
+	var rightFrag: vec4f = texture2D(textureSampler, vUV);
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/anaglyph.fragment.fx:14]:
```suggestion
    var leftFrag: vec4f = textureSample(leftSampler, leftSamplerSampler, input.vUV);
    leftFrag =  vec4f(1.0, leftFrag.g, leftFrag.b, 1.0);

	var rightFrag: vec4f = textureSample(textureSampler, textureSamplerSampler, input.vUV);
```

File: packages/dev/core/src/ShadersWGSL/anaglyph.fragment.fx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+// Samplers
+varying var vUV: vec2f;
+uniform sampler2D textureSampler;
+uniform sampler2D leftSampler;
+
+
+#define CUSTOM_FRAGMENT_DEFINITIONS
+
+fn main(void)
+{
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/anaglyph.fragment.fx:10]:
```suggestion
@fragment
fn main(input: FragmentInputs) -> FragmentOutputs {
```

File: packages/dev/core/src/ShadersWGSL/anaglyph.fragment.fx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+fn main(void)
+{
+    var leftFrag: vec4f = texture2D(leftSampler, vUV);
+    leftFrag =  vec4f(1.0, leftFrag.g, leftFrag.b, 1.0);
+
+	var rightFrag: vec4f = texture2D(textureSampler, vUV);
+    rightFrag =  vec4f(rightFrag.r, 1.0, 1.0, 1.0);
+
+    gl_FragColor =  vec4f(rightFrag.rgb * leftFrag.rgb, 1.0);
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/anaglyph.fragment.fx:17]:
```suggestion
    fragmentOutputs.color =  vec4f(rightFrag.rgb * leftFrag.rgb, 1.0);
```

File: packages/dev/core/src/ShadersWGSL/anaglyph.fragment.fx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+// Samplers
+varying var vUV: vec2f;
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/anaglyph.fragment.fx:2]:
```suggestion
varying vUV: vec2f;
```

File: packages/dev/core/src/ShadersWGSL/depth.vertex.fx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+		#ifdef USE_REVERSE_DEPTHBUFFER
+			vertexOutputs.vDepthMetric = ((-vertexOutputs.position.z + uniforms.depthValues.x) / (uniforms.depthValues.y));
+		#else
+			vertexOutputs.vDepthMetric = ((vertexOutputs.position.z + uniforms.depthValues.x) / (uniforms.depthValues.y));
+		#endif
+	#endif
+
+#if defined(ALPHATEST) || defined(BASIC_RENDER)
+#ifdef UV1
+	vertexOutputs.vUV =  vec2f(uniforms.diffuseMatrix *  vec4f(uvUpdated, 1.0, 0.0));
```
Comment (noname0310) [inline packages/dev/core/src/ShadersWGSL/depth.vertex.fx:69]: construct `vec2f` from `vec4f` seems not valid maybe

**Comment by deltakosh** [comment]
Haha you are faster than me. Anaglyph is not ported yet:)

**Comment by noname0310** [comment]
So the depth.vertex.fx is okay?

**Comment by deltakosh** [comment]
Nope and I need to figure out why my tests did not catch it

I'm still in the process of doing this PR:)

Help appreciated though

**Comment by deltakosh** [comment]
Actually the tests catch it so all good:)

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15398/merge/testResults/webgpuplaywright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15398/merge/testResults/webgpuplaywright/index.html

**Pull request merged**